tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -243,16 +244,44 @@ struct UnsortedSegmentSumFunctor<CPUDevice, T, Index> {
     }
   }
 };
-
+// UnsortedSegmentMaxFunctor implementation for CPUDevice.
+template <typename T, typename Index>
+struct UnsortedSegmentMaxFunctor<CPUDevice, T, Index>
+    : UnsortedSegmentBaseFunctor<CPUDevice, T, Index> {
+  void operator()(OpKernelContext* ctx, const CPUDevice& d,
+                  const Index output_rows, const TensorShape& segment_ids_shape,
+                  typename TTypes<Index>::ConstFlat segment_ids,
+                  const Index data_size, const T* data,
+                  typename TTypes<T, 2>::Tensor output) override {
+    output.setConstant(std::numeric_limits<T>::min());
```

Comment:
I wouldn't expect the GPU and CPU to share implementations, I would expect the CPU implementations of UnsortedSegmentSum/Max to (possibly) share code. Maybe this could be captured as a TODO? (I realize this PR has been pending for quite some time).